README.md

```diff
@@ -31,5 +31,92 @@ This will enable all telemetry events from Red Hat plugins going forward.
 If you want to stop sending usage data to Red Hat, you can disable it in the preferences at Tools > Red Hat Telemetry.
 This will silence all telemetry events from Red Hat plugins going forward.
 
+## How to use from a IDEA plugin extension?
 
+### Dependency
+Start by adding `com.redhat.devtools.intellij.telemetry` to the plugins section of your build.gradle, so that the dependency can be automatically downloaded and installed, when installing your extension from the Marketplace.
+```groovy
+plugins "com.redhat.devtools.intellij.telemetry:0.0.1.9"
+```
 
+### [Optional] Add custom segment keys in src/main/resources/segment.properties
+By default, extensions will send their data to https://app.segment.com/redhat-devtools/sources/intellij/. 
+In development mode, the data is sent to https://app.segment.com/redhat-devtools/sources/intellij_test/.
+You can specify custom segment keys to connect and push usage data to https://segment.com/
+```properties
+"writeKey": "your-segment-key-goes-here",
+"debugWriteKey": "your-segment-key-goes-here-for-dev-mode",
+```
+
+### Create a Telemetry singleton
+You then need to create a singleton that allows you to get hold of the telemetry service.
+Lazy initialisation allows late construction when things are needed.
+
+```java
+public class TelemetryService {
+
+    private static final TelemetryService INSTANCE = new TelemetryService();
+
+    private final Lazy<TelemetryMessageBuilder> builder = new Lazy<>(() -> new TelemetryMessageBuilder(TelemetryService.class.getClassLoader()));
+
+    public static TelemetryMessageBuilder instance() {
+        return INSTANCE.builder.get();
+    }
+}
+```
+### Send a message
+
+To report an event you first need to get a hold on telemetry singleton and start by telling it what action you're about to build.
+Good practice suggests prefixing your action by a group name that you separate with a "-".
```

Comment:
Maybe you can also mention the fact that one cannot use `:` because of the problem with woopra. It can be useful for people who don't know it or even for us in the future so we don't forget